### ISSUE TITLE
Create md5sum file for cg deliver

### ISSUE THREAD

**Issue by keyvanelhami** [Clinical-Genomics/cg#979]
Problem: 
Customers downloading the analysis files from Caesar are unable to check if the downloading worked successfully or not. Therefore md5sum is needed to be created when running `cg deliver` before delivering the files to Caesar.

**Comment by hassanfa** [comment]
Question came up again with another delivery:

1. `cg` should be able to create and write md5 to delivery directory.
2. MD5 is not multithreaded
3. MD5 can only use CPU resources
4. MD5 calculation should and better be submitted as a job to SLURM. Login node cannot handle multiple deliveries.

This can be closed when `cg` delivers MD5 with file delivery.

**Comment by moonso** [comment]
One md5 per file then I suppose?

**Comment by hassanfa** [comment]
Yes. One per file.

**Comment by hassanfa** [comment]
> One md5 per file then I suppose? Or something like https://sourceforge.net/p/rhash/wiki/UsageExamples/

It'd be great if we use a common tool that is available across platforms so customers can replicate results without need. I have not used rhash.

We also have MD5 hash housekeeper's database, maybe we should use that :-)

**Comment by moonso** [comment]
This would be something for @henningonsbring to consider when writing the delivery of Fastq files as well

**Comment by henrikstranneheim** [comment]
@Vince-janv @henningonsbring Is this in place now? Or is just for syncing from clinical-delivery?

**Comment by henningonsbring** [comment]
We only have this for when syncing from clinical-delivery

**Comment by henningonsbring** [comment]
During my time on cg I have only experienced corruption of files once when transferring data to caesar. That time a simple visual inspection of the file made it obvious that it was corrupt. Then we could retrospectively make an md5 sum for what is on hasta, what was uploaded to caesar and what the customer got to identify in which transfer step the file got corrupted. After that we took appropriate measures and fixed the problem.

I think implementing md5-checks for everything we upload to caesar will cause us more pain than gain. I suggest we manually deal with the few corrupted cases retrospectively (and close this issue).
